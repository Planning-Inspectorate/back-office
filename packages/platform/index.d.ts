import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
