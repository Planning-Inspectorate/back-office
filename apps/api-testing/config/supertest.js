// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
