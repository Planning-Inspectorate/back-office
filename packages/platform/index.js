export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
