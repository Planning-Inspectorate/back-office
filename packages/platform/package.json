{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
