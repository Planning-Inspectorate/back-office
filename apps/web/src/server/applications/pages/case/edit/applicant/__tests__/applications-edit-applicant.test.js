import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestApplication } from '../../../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestApplication();
const request = supertest(app);
const successGetResponse = { id: 1, applicants: [{ id: 1, address: { town: 'London' } }] };

const nocks = () => {
	nock('http://test/').get('/applications/case-officer').reply(200, successGetResponse);
	nock('http://test/')
		.get(/\/applications\/123(.*)/g)
		.times(2)
		.reply(200, successGetResponse);
};

describe('applications create applicant', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	beforeEach(async () => {
		await request.get('/applications-service/case-officer');
	});

	describe('GET edit/applicant-organisation-name', () => {
		const baseUrl = '/applications-service/case/123/edit/applicant-organisation-name';

		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Save changes');
		});
	});

	describe('GET edit/applicant-full-name', () => {
		const baseUrl = '/applications-service/case/123/edit/applicant-full-name';

		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Save changes');
		});
	});

	describe('GET edit/applicant-email', () => {
		const baseUrl = '/applications-service/case/123/edit/applicant-email';

		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Save changes');
		});
	});

	describe('GET edit/applicant-website', () => {
		const baseUrl = '/applications-service/case/123/edit/applicant-website';

		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Save changes');
		});
	});

	describe('GET edit/applicant-telephone-number', () => {
		const baseUrl = '/applications-service/case/123/edit/applicant-telephone-number';

		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Save changes');
		});
	});

	describe('GET edit/applicant-address', () => {
		beforeEach(async () => {
			nocks();
		});

		it('should render the read only page', async () => {
			const baseUrl = '/applications-service/case/123/edit/applicant-address';

			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Save changes');
			expect(element.innerHTML).toContain('London');
		});

		it('should render the form page', async () => {
			const baseUrl = '/applications-service/case/123/edit/applicant-address/new';

			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Find address');
		});
	});
});
