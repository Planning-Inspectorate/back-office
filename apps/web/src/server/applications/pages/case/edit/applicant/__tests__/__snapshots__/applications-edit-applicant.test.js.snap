// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications create applicant GET edit/applicant-address should render the form page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> Enter the Applicant’s address</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <input type=\\"hidden\\" name='currentFormStage' value=\\"searchPostcode\\">
        <div class=\\"govuk-form-group\\">
            <h1 class=\\"govuk-label-wrapper\\"><label class=\\"govuk-label font-weight--700\\" for=\\"postcode\\"> Postcode</label></h1>
            <input class=\\"govuk-input govuk-!-width-one-third\\" id=\\"postcode\\" name=\\"postcode\\"
            type=\\"text\\" value=\\"\\">
        </div>
        <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Find address</button>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\"></div>
        </div>
    </form>
</main>"
`;

exports[`applications create applicant GET edit/applicant-address should render the read only page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> Enter the Applicant’s address</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <p class=\\"govuk-body\\"></p>
        </div>
        <input type=\\"hidden\\" name=\\"postcode\\" value=\\"\\">
        <div class=\\"govuk-form-group\\">
            <p class=\\"govuk-label font-weight--700\\">Postcode</p>
            <div class=\\"govuk-grid-row\\">
                <div class=\\"govuk-grid-column-one-third\\">
                    <div class=\\"govuk-grid-row govuk-button-group\\">
                        <p class='govuk-body govuk-grid-column-one-half'></p>
                        <div class=\\"govuk-grid-column-one-half\\"><a class=\\"govuk-link \\" href=\\"applicant-address/new\\">Change</a>
                        </div>
                    </div>
                </div>
            </div>
        </div>
        <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\"></div>
        </div>
    </form>
</main>"
`;

exports[`applications create applicant GET edit/applicant-email should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> Enter the Applicant’s email address</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <div id=\\"applicant.email-hint\\" class=\\"govuk-hint\\">for example, contact@buildinganewbridge.com</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"applicant.email\\" name=\\"applicant.email\\" type=\\"text\\" aria-describedby=\\"applicant.email-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications create applicant GET edit/applicant-full-name should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> Enter the Applicant’s contact name</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicant.firstName\\">First name</label>
            <input class=\\"govuk-input govuk-!-width-one-third\\" id=\\"applicant.firstName\\"
            name=\\"applicant.firstName\\" type=\\"text\\">
        </div>
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicant.lastName\\">Last name</label>
            <input class=\\"govuk-input govuk-!-width-one-third\\" id=\\"applicant.lastName\\"
            name=\\"applicant.lastName\\" type=\\"text\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications create applicant GET edit/applicant-organisation-name should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> Enter the Applicant’s organisation</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <input class=\\"govuk-input govuk-!-width-one-third\\" id=\\"applicant.organisationName\\"
            name=\\"applicant.organisationName\\" type=\\"text\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications create applicant GET edit/applicant-telephone-number should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> Enter the Applicant’s phone number</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <div id=\\"applicant.phoneNumber-hint\\" class=\\"govuk-hint\\">for example, 0300 123 3000</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"applicant.phoneNumber\\" name=\\"applicant.phoneNumber\\" type=\\"text\\" aria-describedby=\\"applicant.phoneNumber-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`applications create applicant GET edit/applicant-website should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> Enter the Applicant’s website</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <div id=\\"applicant.website-hint\\" class=\\"govuk-hint\\">for example, https://www.heathrow.com</div>
            <input class=\\"govuk-input govuk-!-width-one-third\\"
            id=\\"applicant.website\\" name=\\"applicant.website\\" type=\\"text\\" aria-describedby=\\"applicant.website-hint\\">
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save changes</button>
            </div>
        </div>
    </form>
</main>"
`;
